android/build.gradle

```diff
@@ -207,3 +207,80 @@ dependencies {
     implementation "androidx.core:core-ktx:1.6.0"
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
 }
+
+def isGHExampleApp() {
+    return project.hasProperty('isGHExampleApp') && project.property('isGHExampleApp') == "true"
+}
+
+def getAbsoluteCodegenArtifactsPaperDestination() {
+    if (!project.hasProperty('codegenArtifactsPaperDestination')) {
+        throw new Exception('[RNGH] Please fill codegenArtifactsPaperDestination variable in android/gradle.properties correct path to paper paper destination')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsPaperDestination')}"
+}
+
+def getAbsoluteCodegenArtifactsSource() {
+    if (!project.hasProperty('codegenArtifactsSource')) {
+        throw new Exception('[RNGH] Please fill codegenArtifactsSource variable in android/gradle.properties correct path to codegenerated artifacts')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsSource')}"
+}
+
+
+tasks.register("copyCodegenArtifacts") {
+    group 'After build tasks'
+    description 'Tasks which copy codegen artifacts to paper architecture'
+    
+    if (!isGHExampleApp() || !isNewArchitectureEnabled()) {
+        return
+    }
+
+    dependsOn tasks.generateCodegenArtifactsFromSchema
+
+    doLast {
+
+        def absoluteCodegenArtifactsPaperDestination = getAbsoluteCodegenArtifactsPaperDestination()
+        def absoluteCodegenArtifactsSource = getAbsoluteCodegenArtifactsSource()
+
+        def existingFiles = fileTree(absoluteCodegenArtifactsPaperDestination).matching {
+            include '**/*.java'
+        }
+
+        def generatedFiles = fileTree(absoluteCodegenArtifactsSource).matching {
+            include '**/*.java'
+        }
+
+        def existingFilesMap = [:]
+
+        existingFiles.forEach { existingFile ->
+            println existingFile
+            existingFilesMap[existingFile.name] = 1
+        }
+
+        generatedFiles.forEach { generatedFile -> 
+            if (!existingFilesMap.containsKey(generatedFile.name)) {
+                logger.warn("[RNGH] ${generatedFile.name} not found in paper dir, if it's used in Android you need to copy it manually and implement yourself before using auto-copy feature")
+            }
+        }
+
+        if (existingFiles.size() == 0) {
+            logger.warn("[RNGH] Paper destination with codegen interfaces is empty. This might be okay if you don't have any interfaces/delegates used in Android, if that's not the case please check if codegenArtifactsPaperDestination in android/gradle.properties is correct")
+        }
+
+        if (existingFiles.size() > generatedFiles.size()) {
+            throw new Exception("[RNGH] Number od generated artifacts should be greather then or equal to paper interfaces. Please check if codegenArtifactsSource in android/gradle.properties is correct")
```

Comment:
```suggestion
            throw new Exception("[react-native-gesture-handler] Number od generated artifacts should be greater than or equal to paper interfaces. Please check if codegenArtifactsSource in android/gradle.properties is correct or resolve it manually")
```

Reply:
Changed without `or resolve it manually` in https://github.com/software-mansion/react-native-gesture-handler/pull/2933/commits/d5eb15994285ba3dce2185c05c8f1755f6c4959f. I don't think `resolve manually` make sense here - you can have more codegen interfaces (for example for iOS), but imo it shouldn't be less then those for paper

Reply:
Yeah, that sounds good

Reply:
Changed in: https://github.com/software-mansion/react-native-gesture-handler/pull/2933/commits/7e4da8d6e5dcc10337a96a60d5e4c44c6d75283b